### PR TITLE
OpenKE Pytorch compatibility with link prediction scores

### PR DESCRIPTION
**Pull request by Charanyash**

### CODE REVIEW

**Review by svkeerthy — CHANGES_REQUESTED** [review]
Can you please rename Openke directory to OpenKE? It will help in better diffing.

**Comment by svkeerthy** [comment]
Please fix the formatting to address pre-commit check failure. You may need to install the [pre-commit package](https://pre-commit.com/) and setup the hooks given [here](https://github.com/IITH-Compilers/IR2Vec/blob/main/.pre-commit-config.yaml).

File: seed_embeddings/OpenKE/LICENSE
Comment (svkeerthy) [inline seed_embeddings/OpenKE/LICENSE:1]: Please retain this license.

File: seed_embeddings/OpenKE/analogies.txt
Comment (svkeerthy) [inline seed_embeddings/OpenKE/analogies.txt:1]: Please retain this file.

File: seed_embeddings/OpenKE/generate_embedding_ray.py [hunk 1]
```diff
@@ -172,21 +206,43 @@ def findRep(src, dest, index_dir):
...
-        tune_config=TuneConfig(max_concurrent_trials=4),
+        tune_config=TuneConfig(
+            max_concurrent_trials=4,
+            scheduler=scheduler,
+            num_samples=8,
+        ),
+        run_config=RunConfig(
+            checkpoint_config=CheckpointConfig(
+                num_to_keep=2,
+                # *Best* checkpoints are determined by these params:
```
Comment (svkeerthy) [inline seed_embeddings/OpenKE/generate_embedding_ray.py:229]: These values are to be set right? We can check if `is_analogy` is set and select based on that.

File: seed_embeddings/OpenKE/generate_embedding_ray.py [hunk 2]
```diff
@@ -172,21 +206,43 @@ def findRep(src, dest, index_dir):
...
+                # checkpoint_score_attribute="loss",
+                # checkpoint_score_order="min",)
+            ),
+        ),
     )
 
+    # print(tuner.run_config)
     results = tuner.fit()
-
+    # print("Best config: ", results.get_best_result(metric="lose", mode="min"))
```
Comment (svkeerthy) [inline seed_embeddings/OpenKE/generate_embedding_ray.py:238]: Results are to be written/printed right?

File: seed_embeddings/OpenKE/preprocess.py
Comment (svkeerthy) [inline seed_embeddings/OpenKE/preprocess.py:1]: This file should be retained.

File: seed_embeddings/OpenKE/openKE.yaml [hunk 3]
```diff
@@ -1,7 +1,75 @@
-name: openKE
+name: openke
```
Comment (svkeerthy) [inline seed_embeddings/OpenKE/openKE.yaml:1]: Please change it to openKE

**Review by svkeerthy — CHANGES_REQUESTED** [review]

File: seed_embeddings/OpenKE/generate_embedding_ray.py (hunk 2 above)
Comment (Charanyash) [reply seed_embeddings/OpenKE/generate_embedding_ray.py:250]: Yes we can do that

File: seed_embeddings/OpenKE/preprocess.py
Comment (Charanyash) [reply seed_embeddings/OpenKE/preprocess.py:1]: Done

File: seed_embeddings/OpenKE/openKE.yaml (hunk 3 above)
Comment (Charanyash) [reply seed_embeddings/OpenKE/openKE.yaml:1]: Okay I will rename it

File: seed_embeddings/OpenKE/analogies.txt
Comment (Charanyash) [reply seed_embeddings/OpenKE/analogies.txt:1]: I did that

File: seed_embeddings/OpenKE/LICENSE
Comment (Charanyash) [reply seed_embeddings/OpenKE/LICENSE:1]: Done

Comment (Charanyash) [reply seed_embeddings/OpenKE/LICENSE:1]: Done

**Comment by Charanyash** [comment]
I have updated the code with requested changes and added flags for link prediction and analogies calculations while training. Please check the latest code

File: seed_embeddings/OpenKE/analogy.py [hunk 4]
```diff
@@ -27,6 +28,32 @@ def findVec(str1, src):
...
+def genSimilarityTableFromDict(vec, entity_dict):
+    """
+    Generates cosine and Euclidean similarity tables based on the entity embeddings dictionary.
+    """
+    cosineDict = {}
+    euclDict = {}
+
+    for opcode, value in entity_dict.items():
+        value = np.array(value)
+        cosineDict[opcode] = spatial.distance.cosine(vec, value)
```
Comment (svkeerthy) [inline seed_embeddings/OpenKE/analogy.py:51]: Would using pairwise cosine_similarity speed up the computation? Please check. https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.cosine_similarity.html

File: seed_embeddings/OpenKE/analogy.py [hunk 5]
```diff
@@ -27,6 +28,32 @@ def findVec(str1, src):
...
+        euclDict[opcode] = spatial.distance.euclidean(vec, value)
```
Comment (svkeerthy) [inline seed_embeddings/OpenKE/analogy.py:52]: Can be commented out for now.

File: seed_embeddings/OpenKE/config/Trainer.py [hunk 6]
```diff
@@ -148,6 +197,12 @@ def run(self, link_prediction=False, test_dataloader=None, model=None, ray=True)
                     os.path.join(self.checkpoint_dir + "-" + str(epoch) + ".ckpt")
                 )
 
+        print("out_path : ", self.out_path)
+        if self.out_path:
+            print("Inside out_path")
```
Comment (svkeerthy) [inline seed_embeddings/OpenKE/config/Trainer.py:202]: See that these logs are cleaned up.

File: seed_embeddings/OpenKE/data/TrainDataLoader.py [hunk 7]
```diff
@@ -126,6 +127,7 @@ def read(self):
         self.batch_t_addr = self.batch_t.__array_interface__["data"][0]
         self.batch_r_addr = self.batch_r.__array_interface__["data"][0]
         self.batch_y_addr = self.batch_y.__array_interface__["data"][0]
+        print("END OF READ")
```
Comment (svkeerthy) [inline seed_embeddings/OpenKE/data/TrainDataLoader.py:130]: Pl remove this.

File: seed_embeddings/OpenKE/generate_embedding_ray.py [hunk 8]
```diff
@@ -66,54 +66,61 @@ def train(arg_conf):
...
+    print("After test_dataloader")
     transe = TransE(
         ent_tot=train_dataloader.get_ent_tot(),
         rel_tot=train_dataloader.get_rel_tot(),
         dim=arg_conf["dim"],
         p_norm=1,
         norm_flag=True,
     )
-
+    print("After Transe")
```
Comment (svkeerthy) [inline seed_embeddings/OpenKE/generate_embedding_ray.py:82]: Pl remove debug logs.

File: seed_embeddings/OpenKE/generate_embedding_ray.py [hunk 9]
```diff
@@ -206,46 +222,113 @@ def findRep(src, dest, index_dir):
...
-    # print("Best config: ", results.get_best_result(metric="lose", mode="min"))
+
     # Write the best result to a file, best_result.txt
-    # with open(os.path.join(search_space["index_dir"], "best_result.txt"), "a") as f:
-    #     f.write(
-    #         "\n" + str(results.get_best_result(metric="AnalogiesScore", mode="max"))
-    #     )
 
-    # print("Best config: ", results.get_best_result(metric="AnalogiesScore", mode="max"))
+    if arg_conf.is_analogy:
```
Comment (svkeerthy) [inline seed_embeddings/OpenKE/generate_embedding_ray.py:308]: Save the final embeddings corresponding to the best model.

**Review by svkeerthy — CHANGES_REQUESTED** [review]

**Review by svkeerthy — CHANGES_REQUESTED** [review]
Please update the README

**Review by svkeerthy — APPROVED** [review]

**Pull request merged**